### ISSUE TITLE
Should assignments be excluded from undesirable_function_linter?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2815]
```r
lint(text="foo <- 2", linters=undesirable_function_linter(c(foo=NA)))
# <text>:1:1: style: [undesirable_function_linter] Avoid undesirable function "foo".
# foo <- 2
# ^~~
```

That looks like a false positive to me. If anything, that should be covered by `object_overwrite_linter()` if the user wants to avoid masking `foo`.

**Comment by Bisaloo** [comment]
I would go a bit further and then that `undesirable_function_linter()` should be restricted to function calls.

If we only ignore assignments and I define a `foo` variable as you did, I cannot then use it without false positives.

``` r
lintr::lint(
  text = 'paste0(foo, "bar")',
  linters = lintr::undesirable_function_linter(c(foo = NA))
)
#> <text>:1:8: style: [undesirable_function_linter] Avoid undesirable function "foo".
#> paste0(foo, "bar")
#>        ^~~
```

<sup>Created on 2025-07-21 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
we can toggle that with the argument symbol_is_undesirable
